server/src/main/java/org/opensearch/extensions/ExtensionRequest.java

```diff
@@ -24,41 +25,60 @@
  */
 public class ExtensionRequest extends TransportRequest {
     private static final Logger logger = LogManager.getLogger(ExtensionRequest.class);
-    private ExtensionsManager.RequestType requestType;
+    private final ExtensionsManager.RequestType requestType;
+    private final String uniqueId;
 
     public ExtensionRequest(ExtensionsManager.RequestType requestType) {
+        this(requestType, null);
+    }
+
+    public ExtensionRequest(ExtensionsManager.RequestType requestType, String uniqueId) {
         this.requestType = requestType;
```

Comment:
Is there a reason that we don't have a `super` call here, when we have one in the `StreamInput` constructor?

Reply:
you means the constructor only for `RequestType`?

Reply:
No the one with both fields

Reply:
I guess if the super constructor only has the one argument, then the super call should be in the other constructor actually.

Reply:
For here, I am trying to use constructor overloading. we can define the first constructor is a special case of the second constructor, constructor can uses the `this` keyword to call another constructor, and for the first constructor we only need `requestType` so we set the argument `uniqueId` as `null` and only pass `requestType`.

Reply:
Yes, I understand why you're using overloading here, but that does not mean that we can't also use `super(requestType)` before the call to the second constructor.

Reply:
I looked more into it, and there isn't a constructor on `TransportRequest` that only uses requestType so we're fine in this case.  In general though, even when overloading we can still use a `super` call if the same method exists in the class that we are extending.

Reply:
You are right, I will delete the first constructor.

Reply:
Don't delete it - the overload is necessary here

Reply:
Everything is good with the constructors on this PR